src/VecSim/algorithms/hnsw/hnsw_batch_iterator.h

```diff
@@ -78,7 +83,7 @@ HNSW_BatchIterator<DataType, DistType>::HNSW_BatchIterator(
 
     this->dist_func = index->getDistFunc();
     this->dim = index->getDim();
-    this->entry_point = index->getEntryPointId();
+    this->entry_point = index->safeGetEntryPointCopy();
```

Comment:
Can entering from an out-of-date entry point can cause issues? maybe we should call `safeGetEntryPointCopy` just before using its value?